### DISCUSSION TITLE
Пересчет полей в расчетной таблице в цикле при удалении строк в простой таблице

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#576] [Как это сделать?]
Всем добрый день. 

Помогите разобраться со следующим вопросом. Есть несколько полей в расчетной таблице в цикле, значения в которых должны пересчитываться при изменении значений в полях простой таблицы и при удалении строк в этой же таблице.

Строки в простую таблицу добавляются через код действия кнопки (insertList), которая расположена в цикле (при нажатии на данную кнопку в таблицу в самом цикле так же добавляется 1 строка, в которую подставляется определенный порядковый номер). 

При добавлении строк в простую таблицу в каждую строку подставляется номер цикла и порядковый номер, который соответствует строке в расч. таблице цикла, ну и остальные значения в другие поля. Алексей с целью оптимизации советовал делать recalc не через код действия полей в таблице, а через код действия самой таблицы, т.к. одновременно измененных значений в полях может быть множество и так же одновременно может быть удалено множество строк.

Как получить id строк и name полей при изменении в коде действия простой таблицы: тут все понятно - есть переменная $#changes, с помощью которой получаем все id строк, в которых произошли изменения, находим уникальные порядковые номера (которые соответствуют номерам в строках в циклах) по id и name полей и скармливаем все это в recalc, НО ПРИ УДАЛЕНИИ совсем не понятно, как получить тот самый порядковый номер, который был удален... При удалении changes выдает только список id удаленных строк и как получить onfv порядковых номер - совсем не пойму, а может это и вовсе не возможно и тут нужен другой подход. В общем, подскажите, каким методом лучше решать вопрос с пересчетом при удалении в данной ситуации.

**Comment by ttm-alexeypolunin** [comment]
$#changes не содержит значений — это мой пробел в знаниях. Напишу чуть позже как это сделать через глобальные переменные.
